storage/elasticsearch/client.go

```diff
@@ -282,70 +362,184 @@ func (c *ElasticSearchClient) FormatFilter(filter *filters.Filter, mapKey string
 	return nil
 }
 
+func (c *ElasticSearchClient) shouldRollIndexByCount() bool {
+	if c.index.entriesLimit == -1 {
+		logging.GetLogger().Debugf("%s entries limit not set", c.index.name)
+		return false
+	}
+	logging.GetLogger().Debugf("%s entries counter is %d", c.index.name, c.index.entriesCounter)
+	if c.index.entriesCounter < c.index.entriesLimit {
+		return false
+	}
+	c.indexer.Flush()
+	time.Sleep(3 * time.Millisecond)
+
+	c.index.entriesCounter = c.countEntries()
+	if c.index.entriesCounter < c.index.entriesLimit {
+		return false
+	}
+	logging.GetLogger().Debugf("%s enough entries to roll", c.index.name)
+	return true
+}
+
+func (c *ElasticSearchClient) shouldRollIndexByAge() bool {
+	if c.index.ageLimit == -1 {
+		logging.GetLogger().Debugf("%s age limit not set", c.index.name)
+		return false
+	}
+	age := int(time.Now().Sub(c.index.timeCreated).Seconds())
+	logging.GetLogger().Debugf("%s age is %d", c.index.name, age)
+	if age < c.index.ageLimit {
+		return false
+	}
+	logging.GetLogger().Debugf("%s old enough to roll", c.index.name)
+	return true
+}
+
+func (c *ElasticSearchClient) shouldRollIndex() bool {
+	return (c.shouldRollIndexByCount() || c.shouldRollIndexByAge())
+}
+
+func (c *ElasticSearchClient) delIndices() {
+	if c.index.indicesLimit == -1 {
+		logging.GetLogger().Debugf("No indices limit specified for %s", c.index.name)
+		return
+	}
+
+	indices := c.connection.GetCatIndexInfo(c.GetIndexAlias() + "_*")
+	sort.Slice(indices, func(i, j int) bool {
+	  return indices[i].Name < indices[j].Name
+	})
+
+	numToDel := len(indices) - c.index.indicesLimit
+	if numToDel <= 0 {
+		return
+	}
+
+	for _, esIndex := range indices[:numToDel] {
+		logging.GetLogger().Debugf("Deleting index of %s: %s", c.index.name, esIndex.Name)
+		if _, err := c.connection.DeleteIndex(esIndex.Name); err != nil {
+			logging.GetLogger().Errorf("Error deleting index %s: %s", esIndex.Name, err.Error())
+		}
+	}
+}
+
+func (c *ElasticSearchClient) RollIndex() error {
+	c.indexer.Flush()
+	time.Sleep(3 * time.Millisecond)
```

Comment:
If flush always requires a delay then add it to Flush or add a utility function which does both Flush and Sleep.